geps/gep-2162.md

```diff
@@ -0,0 +1,108 @@
+# GEP-2162: Supported features in GatewayClass Status
+
+* Issue: [#2162](https://github.com/kubernetes-sigs/gateway-api/issues/2162)
+* Status: Provisional
+
+## TLDR
+
+This GEP proposes to enhance the [GatewayClassStatus](https://github.com/kubernetes-sigs/gateway-api/blob/f2cd9bb92b4ff392416c40d6148ff7f76b30e649/apis/v1beta1/gatewayclass_types.go#L185) to include a list of Gateway API features supported by the installed GatewayClass. 
+
+## Goals
+
+* Improve UX by enabling users to easily see what features the implementation (GatewayClass) support.
+
+* Automatically run conformance tests based on the supported features populated in GatewayClass status.
+
+* Provide foundation for tools to block or warn when unsupported features are used.
+
+## Non-Goals
+
+* Validate correctness of supported features published by the implementation. Meaning we don't intend to verify whether the supported features reported by the implementation are indeed supported.
+
+## Introduction
+
+The current [GatewayClassStatus](https://github.com/kubernetes-sigs/gateway-api/blob/f2cd9bb92b4ff392416c40d6148ff7f76b30e649/apis/v1beta1/gatewayclass_types.go#L185 is only used to store conditions the controller publishes.
+
+Partnered with the [Conformance Profiles](https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-1709.md) work, we want to:
+
+1. Improve UX by enabling users to easily see what features the implementation(GatewayClass) support.
+1. Automatically run conformance tests based on the supported features populated in GatewayClass status.
+1. Potentially build tooling to block or warn when unsupported features are used (more under [Future Work](#future-work)).
+
+This doc proposes to enhance the GatewayClassStatus API so implementations could publish a list of features they support/don't support.
+
+Implementations **must** publish the supported features before Accepting the GatewayClass.
```

Comment:
Maybe this? As written, this demands two GatewayClass updates, rather than one.
```suggestion
Implementations **must** publish the supported features before Accepting the GatewayClass, or in the same operation.
```